orthoinference/src/main/java/org/reactome/orthoinference/OrthologousEntityGenerator.java

```diff
@@ -43,71 +43,70 @@ public static GKInstance createOrthoEntity(GKInstance entityInst, boolean overri
-				// to be inferred since they wouldn't change between species {Note from infer_events.pl -- David Croft}.
-				} else if (entityInst.getSchemClass().isa(SimpleEntity))
-				{
-					logger.info("PE is a SimplyEntity, using original instance");
-					infEntityInst = entityInst;
-				} else {
-					logger.warn("Unknown PhysicalEntity class: " + entityInst.getClass());
-				}
+				infEntityInst = createInfEWAS(entityInst, override);
+			} else {
 				if (override)
 				{
-					return infEntityInst;
+					logger.info("Mock GEE instance needed");
+					GKInstance mockedInst = InstanceUtilities.createMockGKInstance(entityInst);
+					return mockedInst;
 				}
-				orthologousEntityIdenticals.put(entityInst, infEntityInst);
+			}
+		// Infers Complex or Polymer instances -- Will recursively call createOrthoEntity with override on its constituent PEs
+		} else if (entityInst.getSchemClass().isa(Complex) || entityInst.getSchemClass().isa(Polymer))
+		{
+			infEntityInst = createInfComplexPolymer(entityInst, override);
+		// Infers EntitySetInstances that themselves contain the species attribute (Not just constituent instances as when hasSpecies is called above),
+		// returning the current instance if it doesn't.
+		} else if (entityInst.getSchemClass().isa(EntitySet))
+		{
+			if (entityInst.getAttributeValue(species) != null)
+			{
+				infEntityInst = createInfEntitySet(entityInst, override);
 			} else {
-				logger.info("Inferred PE instance already exists");
+				logger.info("EntitySet has no species attribute, using original instance: " + entityInst);
+				infEntityInst = entityInst;
 			}
-			logger.info("PE inference completed: " + entityInst);
-			return orthologousEntityIdenticals.get(entityInst);
+		// Handles SimpleEntities by returning the current instance. The idea behind this is that SimpleEntities wouldn't need
+		// to be inferred since they wouldn't change between species {Note from infer_events.pl -- David Croft}.
+		} else if (entityInst.getSchemClass().isa(SimpleEntity))
+		{
+			logger.info("PE is a SimplyEntity, using original instance");
+			infEntityInst = entityInst;
 		} else {
-			// This used to have a conditional statement based on the returned value of the 'check_intracellular' function.
-			// That function doesn't exist anymore (only seemed to apply to the 'mtub' species, which hasn't been inferred for a while).
-			// Since the instance is species-agnostic, just returns the original instance.
-			logger.info("Could not find valid species attribute, returning original instance: " + entityInst);
-			return entityInst;
+			logger.warn("Unknown PhysicalEntity class: " + entityInst.getClass());
+		}
+		if (override)
+		{
+			return infEntityInst;
 		}
+		orthologousEntityIdenticals.put(entityInst, infEntityInst);
+			logger.info("PE inference completed: " + entityInst);
+			return orthologousEntityIdenticals.get(entityInst);
```

Comment:
Since `infEntityInst` was put into the`orthologousEntityIdenticals` map, it might be better just to return `infEntityInst` for clarity and to avoid a lookup.